src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java

```diff
@@ -1,142 +1,212 @@
-	 *
-	 * @param chassisSpeeds Chassis Speeds to set.
-	 */
-	public void setChassisSpeeds(ChassisSpeeds chassisSpeeds) {
-		swerveDrive.setChassisSpeeds(chassisSpeeds);
+	public void driveRobotRelative(ChassisSpeeds speeds) {
+		ChassisSpeeds targetSpeeds = new ChassisSpeeds(-speeds.vxMetersPerSecond,-speeds.vyMetersPerSecond,speeds.omegaRadiansPerSecond);
+
+		SwerveModuleState[] states = kinematics.toSwerveModuleStates(targetSpeeds);
+		for (int i = 0; i < modules.length; i++) {
+			modules[i].setState(states[i]);
+		}
 	}
 
-    /**
-	 * Gets the current yaw angle of the robot, as reported by the swerve pose
-	 * estimator in the underlying drivebase.
-	 * Note, this is not the raw gyro reading, this may be corrected from calls to
-	 * resetOdometry().
-	 *
-	 * @return The yaw angle
-	 */
-
-	public ChassisSpeeds getRobotVelocity() {
-		return swerveDrive.getRobotVelocity();
+	public Pose2d getCurrentPose() {
+		return POSE_ESTIMATOR.getEstimatedPose();
 	}
 
-	public void setupPathPlanner() {
-		AutoBuilder.configureHolonomic(
-				this::getPose, // Robot pose supplier
-				this::resetOdometry, // Method to reset odometry (will be called if your auto has a starting pose)
-				this::getRobotVelocity, // ChassisSpeeds supplier. MUST BE ROBOT RELATIVE
-				this::setChassisSpeeds, // Method that will drive the robot given ROBOT RELATIVE ChassisSpeeds
-				new HolonomicPathFollowerConfig(
-						new PIDConstants(0, 0.1, 0.1), // Translation PID constants
-						new PIDConstants(0.01, 0.0, 0.001), // Rotation PID constants
+	public void resetOdom(Pose2d pose) {
+		POSE_ESTIMATOR.resetOdomGivenPose2d(pose);
+	  }
+
+	public void drive(double xVelocity, double yVelocity, double rotationalVelocity) {
+		ChassisSpeeds speeds =
+				ChassisSpeeds.fromFieldRelativeSpeeds(xVelocity, yVelocity, rotationalVelocity, POSE_ESTIMATOR.getEstimatedPose().getRotation());
+	
+		speeds = ChassisSpeeds.discretize(speeds, dtSeconds);
+		SwerveModuleState[] states = kinematics.toSwerveModuleStates(speeds);
+		for (int i = 0; i < modules.length; i++) {
+			modules[i].setState(states[i]);
+		}
+	}
+	/*
+	* The two commands below are commands to follow a single path for two different "phases", teleop and auto.
+	* The command that I tested was the teleop one, but to actually get the right velocities to use them with a pigeon and gyro
+	*  is too much work, so I made a different command.
+	* followPathCommandTeleop accepts an x,y, and rotationalVelocity that I tried to get using the Pigeon, but it doesn't seem to
+	* be accurate. So I created a new getChassisSpeedsAuto command that passes in module states (getModuleStates())
+	* in place of the getChassisSpeeds method that takes in the velocities. Testing is pending, but I just thought that it might be useful.
+	*
+	* AutoBuilder has also been configured in the SwerveSubsystem constructor class, so we can work with autos now.
+	* */
+	
+
+	public Command followPathCommandAuto(String pathName) {
```

Comment:
Is this being used?

Reply:
This is when you only want to run the path instead of running the whole auto I think.